src/prognostic_equations/advection.jl

```diff
@@ -422,6 +427,284 @@ function edmfx_sgs_vertical_advection_tendency!(
+            va = vertical_advection(ᶠwᵢ³ʲs, ᶜmseᵢ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            va = vertical_advection(ᶠwᵣ³ʲs, ᶜmseᵣ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            va = vertical_advection(ᶠwₛ³ʲs, ᶜmseₛ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            #=
+                        va = vertical_advection(ᶠwₗ³ʲs, Y.c.sgsʲs.:($j).q_liq, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+                        va = vertical_advection(ᶠwᵢ³ʲs, Y.c.sgsʲs.:($j).q_ice, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+                        va = vertical_advection(ᶠwᵣ³ʲs, Y.c.sgsʲs.:($j).q_rai, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+                        va = vertical_advection(ᶠwₛ³ʲs, Y.c.sgsʲs.:($j).q_sno, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+            =#
+
+            # mse, q_tot and moisture tracers terms proportional to 1/ρ̂ ∂zρ̂
+            ᶜinv_ρ̂ = (@. lazy(
+                specific(
+                    FT(1),
+                    Y.c.sgsʲs.:($$j).ρa,
+                    FT(0),
+                    Y.c.ρ,
+                    turbconv_model,
+                ),
+            ))
+            ᶜ∂ρ̂∂zₗ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwₗʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+            ᶜ∂ρ̂∂zᵢ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwᵢʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+            ᶜ∂ρ̂∂zᵣ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwᵣʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+            ᶜ∂ρ̂∂zₛ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwₛʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+
+            @. Yₜ.c.sgsʲs.:($$j).q_tot -=
+                dot(
+                    ᶜinv_ρ̂ * ᶜ∂ρ̂∂zₗ,
+                    CT3(Geometry.WVector(-1 * ᶜwₗʲs.:($$j))),
+                ) *
+                Y.c.sgsʲs.:($$j).q_liq *
+                (1 - Y.c.sgsʲs.:($$j).q_tot)
+            @. Yₜ.c.sgsʲs.:($$j).q_liq -=
+                dot(
+                    ᶜinv_ρ̂ * ᶜ∂ρ̂∂zₗ,
+                    CT3(Geometry.WVector(-1 * ᶜwₗʲs.:($$j))),
+                ) * Y.c.sgsʲs.:($$j).q_liq
+            @. Yₜ.c.sgsʲs.:($$j).mse -=
+                dot(
+                    ᶜinv_ρ̂ * ᶜ∂ρ̂∂zₗ,
+                    CT3(Geometry.WVector(-1 * ᶜwₗʲs.:($$j))),
+                ) * (Y.c.sgsʲs.:($$j).q_liq * Y.c.sgsʲs.:($$j).mse - ᶜmseₗ)
```

Comment:
```suggestion
            @. Yₜ.c.sgsʲs.:($$j).mse -=
                dot(
                    ᶜinv_ρ̂ * ᶜ∂ρ̂∂zₗ,
                    CT3(Geometry.WVector(-1 * ᶜwₗʲs.:($$j))),
                ) * (ᶜmseₗ - Y.c.sgsʲs.:($$j).q_liq * Y.c.sgsʲs.:($$j).mse)
```
(I think it's a sign error here, but please double check)